pkg/kubelet/kuberuntime/kuberuntime_container_linux.go

```diff
@@ -298,3 +303,72 @@ func toKubeContainerResources(statusResources *runtimeapi.ContainerResources) *k
 	}
 	return cStatusResources
 }
+
+var isCgroup2UnifiedMode = func() bool {
+	return libcontainercgroups.IsCgroup2UnifiedMode()
+}
+
+type swapConfigurationHelper struct {
+	isCgroup2UnifiedMode bool
```

Comment:
Great suggestion! It would be much clearer that way.
Thanks! done.

Reply:
Not sure I understand.
Can you clarify? getting rid of what, the cached bool variable?

I've cached it since cgroup versions will not change on the node during the runtime of Kubelet. Why do you think it's necessary to remove it?

Reply:
because libcontainer already does the same https://github.com/opencontainers/runc/blob/main/libcontainer/cgroups/utils.go#L34

Reply:
Makes sense! Thanks a lot!
Done.